react/week2/todo/src/components/counter.js

```diff
@@ -0,0 +1,18 @@
+import React, { useState, useEffect } from "react";
+const Counter = () => {
+     const [count, setCount] = useState(0);
+     let t;
+  useEffect(() => {
+    t = setInterval(() => {
+      setCount(count + 1);
+    }, 1000);
+
+    return () => {
+      if (t) {
+        clearInterval(t);
+      }
+    };
+  }, [count]);
```

Comment:
`let t` should be defined inside the function in `useEffect`, or it may give strange sideeffects